packages/example/src/components/Connect/Connect.tsx

```diff
@@ -1,18 +1,21 @@
-import React, { useCallback, useState } from 'react';
-import { ReactComponent as ConnectIcon } from "./image/connect.svg"
-import { ReactComponent as  MetaMaskIcon } from "./image/MetaMask.svg"
-import ArrowRight from "../Icons/ArrowRight"
+import React, { useCallback } from 'react';
+import { ReactComponent as ConnectIcon } from "./image/connect.svg";
+import { ReactComponent as MetaMaskIcon } from "./image/MetaMask.svg";
+import { ReactComponent as LoadingIcon } from "./image/loading.svg";
+import ArrowRight from "../Icons/ArrowRight";
 import Modal from "./Modal";
 import { connect } from "../../lib/snap";
 import { trackConnectClick, trackConnectSucceed } from "../../tracking";
+import { useKeystoneStore } from '../../mobx';
 
 interface ConnectProps {
   open: boolean;
+  close: () => void;
   onConnected: () => void;
 }
 
-const Connect = ({open, onConnected}: ConnectProps) => {
-  const [isConnecting, setIsConnecting] = useState<boolean>(false);
+const Connect = ({open, close, onConnected}: ConnectProps) => {
+  const { runtime: {isConnecting, setIsConnecting} } = useKeystoneStore();
```

Comment:
`isConnecting` is a state which only cares in current component, may be it's not a good idea to put it in global.